[doc] Update nRF Connect - Android commissioning guide (#74)
Damian-Nordic/connectedhomeip: docs/guides/nrfconnect_android_commissioning.md

-   [Overview](#overview)
-   [Requirements](#requirements)
-   [Setting up Thread Border Router](#setting-up-thread-border-router)

> -   [Building and programming nRF Connect Lock Example Application](#building-example)

Reviewer: What about changing Lock Example to Lighting Example or list them both (if not in the title, maybe in the section below).

Author: There is information at the beginning of the article that it can be adapted to the light example, too. I thought we wanted the article to be more specific and have less digressions. Why would the lighting example be better?

Reviewer: That's fine, my thinking was that the current Android Application is better suited for lighting app, as it has also the scroll to control brightness. That's why I was considering making Lighting example primaty, but I'm fine either way.

The commissioning process is composed of the following main stages:

1.  CHIPTool discovers a CHIP accessory device over Bluetooth LE.
2.  CHIPTool establishes a secure channel to the device over Bluetooth LE, and
    sends CHIP operational credentials and Thread provisioning data.
3.  The accessory device joins a CHIP-enabled Thread network.

Bluetooth LE is only used during the commissioning phase. Afterwards, only the

> IPv6 connectivity between the smartphone and the accessory device is needed to
> send operational messages. The IPv6 address of the device is not exchanged
> during the commissioning process and CHIPTool must use DNS Service Discovery
> (DNS-SD) to learn or refresh the address before the IPv6-based communication can
> be initiated.

Reviewer 2: ```suggestion
IPv6 connectivity between the smartphone and the accessory device is needed to
send operational messages. The IPv6 address of the device is not exchanged
during the commissioning process and CHIPTool must use DNS Service Discovery
(DNS-SD) to learn or refresh the address before the controller initiates the IPv6-based communication.
```

> Since a typical smartphone does not have a Thread radio built-in, extra effort
> is needed to prepare a fully-fledged testing environment. This page refers to
> the [OpenThread Border Router](openthread_border_router_pi.md) article to set up
> a Thread Border Router on a Raspberry Pi.

Reviewer 2: ```suggestion
Since a typical smartphone does not have a Thread radio, you need to prepare a fully-fledged testing environment. Refer to
the [OpenThread Border Router](openthread_border_router_pi.md) article to set up
a Thread Border Router on a Raspberry Pi.
```

The following diagram shows the connectivity between network components required
to allow communication between devices running the CHIPTool and Lock
applications:

<img src="../images/nrfconnect_android_connectivity.svg" alt="CHIP nodes connectivity" width="600" />

<hr>

<a name="requirements"></a>

## Requirements

> You need the following hardware and software to go through all the steps
> presented in this article:

Reviewer 2: ```suggestion
You need the following hardware and software for commissioning nRF Connect Accessory using Android CHIPTool:
```

-   Two nRF52840 DK (PCA10056)

>     -   One nRF52840 DK is needed to run

Reviewer 2: ```suggestion
    -   One nRF52840 DK for running the
```

        [OpenThread Radio Co-Processor](https://openthread.io/platforms/co-processor)
        firmware and can be replaced with another compatible device like

>         nRF52840 Dongle.

Reviewer 2: ```suggestion
        firmware. You can replace this DK with another compatible device, such as
        the nRF52840 Dongle.

>     -   One nRF52840 DK is needed to run the Lock application and can be
>         replaced with another compatible device like nRF5340 DK.

Reviewer 2: ```suggestion
    -   One nRF52840 DK for running the lock application. You can replace
	this DK with another compatible device, such as the nRF5340 DK.
```

-   Smartphone compatible with Android 8.0 or later
-   Raspberry Pi Model 3B+ or newer (along with an SD card with at least 8 GB of
    memory)
-   Wi-Fi Access Point supporting IPv6 (without the IPv6 Router Advertisement
    Guard enabled on the router)

<hr>

<a name="setting-up-thread-border-router"></a>

## Setting up Thread Border Router

Follow the [OpenThread Border Router](openthread_border_router_pi.md) article to

> set up OpenThread Border Router on a Raspberry Pi, using either nRF52840 DK or
> nRF52840 Dongle as the
> [OpenThread Radio Co-Processor](https://openthread.io/platforms/co-processor).

Reviewer 2: ```suggestion
set up OpenThread Border Router on the Raspberry Pi, with either the nRF52840 DK or
the nRF52840 Dongle acting as the
[OpenThread Radio Co-Processor](https://openthread.io/platforms/co-processor).
```

3.  The accessory device joins a CHIP-enabled Thread network.

Bluetooth LE is only used during the commissioning phase. Afterwards, only the
IPv6 connectivity between the smartphone and the accessory device is needed to
send operational messages. The IPv6 address of the device is not exchanged
during the commissioning process and CHIPTool must use DNS Service Discovery
(DNS-SD) to learn or refresh the address before the IPv6-based communication can
be initiated.

Since a typical smartphone does not have a Thread
radio built-in, extra effort is needed to prepare a fully-fledged testing
environment. This page refers to the
[OpenThread Border Router](openthread_border_router_pi.md) article to set up a
Thread Border Router on a Raspberry Pi.

The following diagram shows the connectivity between network components required
to allow communication between devices running the CHIPTool and Lock
applications:

<img src="../images/nrfconnect_android_connectivity.svg" alt="CHIP nodes connectivity" width="600" />

<hr>

<a name="requirements"></a>

## Requirements

You need the following hardware and software to go through all the steps
presented in this article:

> -   Two nRF52840 DK (PCA10056)

Reviewer: I would mention nRF5340 as well. Thoughts?

Author: Yeah, good point.
